pkg/resources/cloudcontroller/kubevirt.go

```diff
@@ -0,0 +1,149 @@
+			dep.Name = KubeVirtCCMDeploymentName
+			dep.Labels = resources.BaseAppLabels(KubeVirtCCMDeploymentName, nil)
+
+			dep.Spec.Replicas = resources.Int32(1)
+
+			dep.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: resources.BaseAppLabels(KubeVirtCCMDeploymentName, nil),
+			}
+
+			podLabels, err := data.GetPodTemplateLabels(KubeVirtCCMDeploymentName, dep.Spec.Template.Spec.Volumes, nil)
+			if err != nil {
+				return nil, err
+			}
+
+			dep.Spec.Template.ObjectMeta = metav1.ObjectMeta{
+				Labels: podLabels,
+			}
+
+			dep.Spec.Template.Spec.DNSPolicy, dep.Spec.Template.Spec.DNSConfig, err =
+				resources.UserClusterDNSPolicyAndConfig(data)
+			if err != nil {
+				return nil, err
+			}
+
+			dep.Spec.Template.Spec.AutomountServiceAccountToken = pointer.BoolPtr(false)
+
+			openvpnSidecar, err := vpnsidecar.OpenVPNSidecarContainer(data, openvpnClientContainerName)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get openvpn sidecar: %v", err)
+			}
+
+			version, err := getKVVersion(data.Cluster().Spec.Version)
+			if err != nil {
+				return nil, err
+			}
+
+			dep.Spec.Template.Spec.Volumes = append(getVolumes(), corev1.Volume{
+				Name: resources.CloudConfigConfigMapName,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: resources.CloudConfigConfigMapName,
+						},
+					},
+				},
+			})
+
+			dep.Spec.Template.Spec.Containers = []corev1.Container{
+				*openvpnSidecar,
+				{
+					Name:    ccmContainerName,
+					Image:   data.ImageRegistry(resources.RegistryDocker) + "/mfranczy/kubevirt-cloud-controller-manager:v" + version,
+					Command: []string{"/bin/kubevirt-cloud-controller-manager"},
+					Args:    getKVFlags(data),
+					VolumeMounts: append(getVolumeMounts(), corev1.VolumeMount{
+						Name:      resources.CloudConfigConfigMapName,
+						MountPath: "/etc/kubernetes/cloud",
+						ReadOnly:  true,
+					}),
+				},
+			}
+
+			defResourceRequirements := map[string]*corev1.ResourceRequirements{
+				ccmContainerName:    kvResourceRequirements.DeepCopy(),
+				openvpnSidecar.Name: openvpnSidecar.Resources.DeepCopy(),
+			}
+			err = resources.SetResourceRequirements(dep.Spec.Template.Spec.Containers, defResourceRequirements, nil, dep.Annotations)
+			if err != nil {
+				return nil, fmt.Errorf("failed to set resource requirements: %v", err)
+			}
+			return dep, nil
+		}
+	}
+}
+
+func getKVFlags(data *resources.TemplateData) []string {
+	flags := []string{
+		"--kubeconfig=/etc/kubernetes/kubeconfig/kubeconfig",
+		"--cloud-config=/etc/kubernetes/cloud/config",
+		"--cloud-provider=kubevirt",
+	}
+	if data.Cluster().Spec.Features[kubermaticv1.ClusterFeatureCCMClusterName] {
+		flags = append(flags, "--cluster-name", data.Cluster().Name)
+	}
+	return flags
+}
+
+const latestKubeVirtCCMVersion = "0.0.8"
+
+func getKVVersion(version semver.Semver) (string, error) {
+	if version.Minor() < 17 {
```

Comment:
Would be cool to have this field as a constant as well since it represents a configuration value. Something like: 
```go
MinorKubernetesVersion = 17
```

Similar like how we declare k8s version in code.

Reply:
that part of the code is not needed, forgot to remove it (copy-paste)